### PR TITLE
Support PyTorch 1.11.0

### LINKED ISSUE #3366: CI failed due to the error in AllenNLP integration
### Expected behavior

https://github.com/optuna/optuna/actions/runs/1960157156

### Environment

- Optuna version: 3.0.0b0.dev
- Python version: 3.9.0
- OS: 20.04.3
- (Optional) Other libraries and their versions: AllenNLP 2.9.1

### Error messages, stack traces, or logs

```shell
_ test_allennlp_pruning_callback_with_executor[tests/integration_tests/allennlp_tests/example_with_executor_and_pruner_distributed.jsonnet-HyperbandPruner-pruner_kwargs0] _

pruner_class = <class 'optuna.pruners._hyperband.HyperbandPruner'>
pruner_kwargs = {'max_resource': 10, 'min_resource': 3, 'reduction_factor': 5}
input_config_file = 'tests/integration_tests/allennlp_tests/example_with_executor_and_pruner_distributed.jsonnet'

    @pytest.mark.parametrize(
        "pruner_class,pruner_kwargs",
        [
            (
                optuna.pruners.HyperbandPruner,
                {"min_resource": 3, "max_resource": 10, "reduction_factor": 5},
            ),
            (
                optuna.pruners.MedianPruner,
                {"n_startup_trials": 8, "n_warmup_steps": 1, "interval_steps": 3},
            ),
            (optuna.pruners.NopPruner, {}),
            (
                optuna.pruners.PercentilePruner,
                {"percentile": 50.0, "n_startup_trials": 10, "n_warmup_steps": 1, "interval_steps": 3},
            ),
            (
                optuna.pruners.SuccessiveHalvingPruner,
                {"min_resource": 3, "reduction_factor": 5, "min_early_stopping_rate": 1},
            ),
            (
                optuna.pruners.ThresholdPruner,
                {"lower": 0.0, "upper": 1.0, "n_warmup_steps": 3, "interval_steps": 2},
            ),
        ],
    )
    @pytest.mark.parametrize(
        "input_config_file",
        [
            "tests/integration_tests/allennlp_tests/example_with_executor_and_pruner.jsonnet",
            "tests/integration_tests/allennlp_tests/example_with_executor_and_pruner_distributed.jsonnet",  # noqa: E501
        ],
    )
    def test_allennlp_pruning_callback_with_executor(
        pruner_class: Type[optuna.pruners.BasePruner],
        pruner_kwargs: Dict[str, Union[int, float]],
        input_config_file: str,
    ) -> None:
        def run_allennlp_executor(pruner: optuna.pruners.BasePruner) -> None:
            study = optuna.create_study(direction="maximize", pruner=pruner, storage=storage)
            trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
            trial.suggest_float("DROPOUT", 0.0, 0.5)
            executor = optuna.integration.AllenNLPExecutor(trial, input_config_file, serialization_dir)
            executor.run()
    
        with tempfile.TemporaryDirectory() as tmp_dir:
            pruner_name = pruner_class.__name__
            os.mkdir(os.path.join(tmp_dir, pruner_name))
            storage = "sqlite:///" + os.path.join(tmp_dir, pruner_name, "result.db")
            serialization_dir = os.path.join(tmp_dir, pruner_name, "allennlp")
    
            pruner = pruner_class(**pruner_kwargs)  # type: ignore
>           run_allennlp_executor(pruner)

tests/integration_tests/allennlp_tests/test_allennlp.py:391: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/integration_tests/allennlp_tests/test_allennlp.py:382: in run_allennlp_executor
    executor.run()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <optuna.integration.allennlp._executor.AllenNLPExecutor object at 0x7f7d4443ebb0>

    def run(self) -> float:
        """Train a model using AllenNLP."""
        for package_name in self._include_package:
            allennlp.common.util.import_module_and_submodules(package_name)
    
        # Without the following lines, the transformer model construction only takes place in the
        # first trial (which would consume some random numbers), and the cached model will be used
        # in trials afterwards (which would not consume random numbers), leading to inconsistent
        # results between single trial and multiple trials. To make results reproducible in
        # multiple trials, we clear the cache before each trial.
        # TODO(MagiaSN) When AllenNLP has introduced a better API to do this, one should remove
        # these lines and use the new API instead. For example, use the `_clear_caches()` method
        # which will be in the next AllenNLP release after 2.4.0.
        allennlp.common.cached_transformers._model_cache.clear()
        allennlp.common.cached_transformers._tokenizer_cache.clear()
    
        self._set_environment_variables()
        params = allennlp.common.params.Params(self._build_params())
    
        if "distributed" in params:
    
            if OPTUNA_ALLENNLP_DISTRIBUTED_FLAG in os.environ:
                warnings.warn(
                    "Other process may already exists."
                    " If you have trouble, please unset the environment"
                    " variable `OPTUNA_ALLENNLP_USE_DISTRIBUTED`"
                    " and try it again."
                )
    
            os.environ[OPTUNA_ALLENNLP_DISTRIBUTED_FLAG] = "1"
    
        try:
            allennlp.commands.train.train_model(
                params=params,
                serialization_dir=self._serialization_dir,
                file_friendly_logging=self._file_friendly_logging,
                force=self._force,
                include_package=self._include_package,
            )
        except ProcessRaisedException as e:
            if "raise TrialPruned()" in str(e):
                raise TrialPruned()
    
>       metrics = json.load(open(os.path.join(self._serialization_dir, "metrics.json")))
E       FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpawf96wcz/HyperbandPruner/allennlp/metrics.json'

optuna/integration/allennlp/_executor.py:236: FileNotFoundError
```

### Steps to reproduce

See https://github.com/optuna/optuna/actions/runs/1960157156.

### Additional context (optional)

- Maybe related to https://github.com/allenai/allennlp/releases/tag/v2.9.1
- Error happened on distributed configuration

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Resolve a part of tasks in #2981 and resolve #3366. I'm not sure but https://github.com/optuna/optuna/pull/3501 might be also fixed by this PR.

## Description of the changes

- Remove version constraints by https://github.com/optuna/optuna/pull/3353.
- Bump PyTorch to the latest stable 1.11.0.
    - This version drops python 3.6.
- Do not test PyTorch lightning callback
    - Since PyTorch-lightning breaks backward compatibility as described in https://github.com/optuna/optuna/issues/3418, we will not test this by this PR.
- Skip `test_catboost.py::test_catboost_pruning_callback_errors`
- Add a change requested as in https://github.com/optuna/optuna/pull/3353#discussion_r849478251

### CODE REVIEW

**Comment by nzw0301** [comment]
Hmm, catboost tests failed in mac and linux. I'll investigate it.

**Comment by nzw0301** [comment]
Let's skip catboost test as well...

File: .github/workflows/tests-integration.yml [hunk 1]
```diff
@@ -54,44 +54,37 @@ jobs:
         optuna --version
 
         pip install --progress-bar off .[tests]
-        pip install --progress-bar off .[integration] -f https://download.pytorch.org/whl/torch_stable.html
-
-        # TODO(nzw0301): remove the version constraint
-        # when resolving https://github.com/optuna/optuna/issues/3418
-        pip install --progress-bar off -U "setuptools==59.5.0"
+        pip install --progress-bar off .[integration] --extra-index-url https://download.pytorch.org/whl/cpu
```
Comment (himkt) [inline .github/workflows/tests-integration.yml:57]: [note] `-f https://download.pytorch.org/whl/torch_stable.html` is still used in:  (🔗 https://pytorch.org/get-started/previous-versions) but `--extra-index-url https://download.pytorch.org/whl/cpu` is suggested in (🔗 https://deploy-preview-972--pytorch-dot-org-preview.netlify.app/get-started/locally/).

(🔗 https://github.com/pytorch/pytorch.github.io/pull/972)

**Review by himkt — COMMENTED** [review]
Thank you for the PR, looks basically solid.

**Comment by himkt** [comment]
Could you please review this PR as well @keisuke-umezawa?

**Comment by nzw0301** [comment]
update?

**Review by himkt — APPROVED** [review]
Sincerely apologize for the long inactive period, LGTM.
I'll revise #3517 to fix conflicts after the PR.

**Comment by himkt** [comment]
@keisuke-umezawa Let me kindly ping.

**Comment by himkt** [comment]
Let me kindly ping @keisuke-umezawa. :pray:

**Review by keisuke-umezawa — APPROVED** [review]
Sorry for being late. LGTM!

**Comment by nzw0301** [comment]
Thanks!

**Pull request merged**